### PR TITLE
Location fixes

### PR DESCRIPTION
**Pull request by sudo-joseph**
Updates location behavior. 

1. User is promoted to share location on main page. If they agree, location is saved to session var for future use. If not, session var is initialized to default location (Oakland). Fixes issue where site would hang when location permission not granted. 
2. Implemented address geocoding option. User can enter an address and have loc move to new location. Geocoding implemented via geopy + Map Box geocoding service. Let me know if the flyTo annimation is too jumpy and we can replace with a simple refresh. 

Note this service is free for first 200k uses, but becomes a paid service after that. Requires secret key in .env file to function. This is similar pricing structure for base tiles. (Which i should have called out earlier). To date, we have used 6k/200k monthly allowance. 

To do: Need to implement template based on state abbreviation json. Need to update geocoding call to use full state name in request to minimize incorrect results. Need to force user to enter all values in address fields before clicking button.

### CODE REVIEW

File: apps/core/templates/pages/add_report.html [hunk 1]
```diff
@@ -10,17 +10,103 @@
...
+</script>
+
+<!-- <script>
+function sendAddress() {
+  alert('what the fuck')
+  console.log("hello world");
+}
+
+</script> -->
+
```
Comment (ishinan) [inline apps/core/templates/pages/add_report.html]: Let's remove this test function when cleaning up. This is just a reminder.

**Review by ishinan — APPROVED** [review]
Awesome!

File: apps/core/templates/pages/add_report.html (hunk 1 above)
Comment (sudo-joseph) [reply apps/core/templates/pages/add_report.html]: haha good point.

Comment (sudo-joseph) [reply apps/core/templates/pages/add_report.html]: removed

**Pull request merged**